packages/plugins/graphql/server/services/content-api/index.js

```diff
@@ -70,6 +76,13 @@ module.exports = ({ strapi }) => {
 
       // Apply user-defined plugins
       plugins: extension.plugins,
+
+      // Whether to generate artifacts (GraphQL schema, TS types definitions) or not.
+      // By default, we generate artifacts only on development environment
+      shouldGenerateArtifacts: config('generateArtifacts', process.env.NODE_ENV === 'development'),
```

Comment:
Would be great to use strapi.config.get('environment') when we want to know the env instead of the env var directly